redis/commands/search/result.py

```diff
@@ -39,18 +49,22 @@ def __init__(
 
             fields = {}
             if hascontent and res[i + fields_offset] is not None:
-                fields = (
-                    dict(
-                        dict(
-                            zip(
-                                map(to_string, res[i + fields_offset][::2]),
-                                map(to_string, res[i + fields_offset][1::2]),
-                            )
-                        )
-                    )
-                    if hascontent
-                    else {}
-                )
+                keys = map(to_string, res[i + fields_offset][::2])
+                values = res[i + fields_offset][1::2]
+
+                for key, value in zip(keys, values):
+                    if field_encodings is None or key not in field_encodings:
+                        fields[key] = to_string(value)
```

Comment:
This seems like a second line of defaults, to make sure by default we always decode. But the fact that `decode_field` defaults to `True` and `encoding` defaults to `utf8` should achieve the same result, right? And this class is only used from private methods. So I would not keep this logic here, it's too "smart" for what the `Result` class is supposed to do.

Reply:
This condition ensures that we preserve current behavior for cases when a query doesn't specify return fields.

Reply:
Maybe add a comment explaining this?